docs/references/unittest.rst

```diff
@@ -0,0 +1,179 @@
+Unit Test Framework
+========================
+
+.. note::
+	Available in v0.11.0 and later
+
+Overview
+*********
+
+You can write test cases for your tool by defining the ``tests()`` function in your Tool class.
+Test cases defined by this function will be picked up by ``janisdk run-test`` command.
+
+This unit test framework provides several predefined preprocessors to
+transform Janis execution output data in the format that can be tested.
+For example, there is a preprocessor to read the md5 checksum of an output file.
+In addition to the predefined preprocessors, the frameworks also allows users to define and pass their own preprocessors.
+
+
+Define test cases
+-----------------
+You can define multiple test cases per tool.
+For each test case, you can declare multiple expected outputs.
+Mostly, you really only need to define more test cases if they require different input data.
+
+
```

Comment:
Here, it might be worth something like,

> The classes we use here (`:class:janis.TTestCase`, `:class:janis.TTestExpectedOutput` and `:class:janis.TTestPreprocessor` are declared at the bottom of this document)

Reply:
Yep good point. Just added. Thanks!